src/queue.c

```diff
@@ -6592,6 +6606,11 @@ _dispatch_runloop_queue_handle_dispose(dispatch_lane_t dq)
 #elif defined(__linux__)
 	int rc = close(handle);
 	(void)dispatch_assume_zero(rc);
+#elif defined(__unix__)
+	int rc = close((int)(handle & 0xffffffff));
```

Comment:
That seems a strange constant to use.

Reply:
The handle packs both fds of the pipe into a single 64-bit integer -- this plucks out the lower 32-bits.

I'll define some macros to make this a little more obvious.

Reply:
Oh! How ... fun. ;)